qubes_config/global_config/policy_manager.py

```diff
@@ -0,0 +1,113 @@
+# -*- encoding: utf8 -*-
+#
+# The Qubes OS Project, http://www.qubes-os.org
+#
+# Copyright (C) 2022 Marta Marczykowska-Górecka
+#                               <marmarta@invisiblethingslab.com>
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation; either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+"""Class used to manage PolicyClient and do some convenience processing."""
+import subprocess
+from typing import Optional, List, Tuple
+
+from qrexec.policy.admin_client import PolicyClient
+from qrexec.policy.parser import StringPolicy, Rule
+
+class PolicyManager:
+    """
+    Single manager for interacting with Qubes Policy.
+    Should be used as a singleton.
+    """
+    def __init__(self):
+        self.policy_client = PolicyClient()
+        self.policy_disclaimer = """
+# THIS IS AN AUTOMATICALLY GENERATED POLICY FILE.
+# Any changes made manually may be overwritten by Qubes Configuration Tools.
+"""
+
+    def get_conflicting_policy_files(self, service: str,
+                                     own_file: str) -> List[str]:
+        """
+        Get a list of policy files (as str) that apply to the selected service
+        and are before it in load order.
+        :param service: service name
+        :param own_file: name of the config's own file
+        :return: list of file names as str
+        """
+        files = self.policy_client.policy_get_files(service)
+        conflicting_files = []
+        for f in files:
+            if not f:
+                # this is a workaround; if there is no file applicable to the
+                # policy, PolicyClient returns a single empty string.
+                continue
+            if f == own_file:
```

Comment:
As a future extension, you could load rules from other files found here too, and remember from where you loaded the rule. Then, display all the rules, but allow to modify only those in `own_file` (some rules are locked already - could be the same thing). And display a warning only if some aren't understood, not just when other files are present.

This may have a lot of corner cases, so definitely for a future work.